tools/RAiDER/statsPlot.py

```diff
@@ -454,32 +453,47 @@ def __init__(self, filearg, col_name, workdir='./', bbox=None, spacing=1, timein
         self.timeinterval = timeinterval
         self.seasonalinterval = seasonalinterval
         self.stationsongrids = stationsongrids
+        self.cbounds = cbounds
         self.colorpercentile = colorpercentile
         self.verbose = verbose
 
         # create workdir if it doesn't exist
         if not os.path.exists(self.workdir):
             os.mkdir(self.workdir)
 
-        self.createDF()
+        if self.cbounds:
+            self.cbounds = [float(val) for val in self.cbounds.split()]
 
-    def _getExtent(self):  #dataset, spacing=1, userbbox=None
+        self.create_DF()
+
+    def _get_extent(self):  #dataset, spacing=1, userbbox=None
         """ Get the bbox, spacing in deg (by default 1deg), optionally pass user-specified bbox. Output array in WESN degrees """
-        extent = [np.floor(min(self.df['Lon'])-(self.spacing/2)), np.ceil(max(self.df['Lon'])+(self.spacing/2)),
-                  np.floor(min(self.df['Lat'])-(self.spacing/2)), np.ceil(max(self.df['Lat'])+(self.spacing/2))]
+        extent = [np.floor(min(self.df['Lon'])), np.ceil(max(self.df['Lon'])),
+                  np.floor(min(self.df['Lat'])), np.ceil(max(self.df['Lat']))]
         if self.bbox is not None:
             dfextents_poly = Polygon(np.column_stack((np.array([extent[0], extent[0], extent[1], extent[1], extent[0]]),
                                                       np.array([extent[2], extent[3], extent[3], extent[2], extent[2]]))))
             userbbox_poly = Polygon(np.column_stack((np.array([self.bbox[2], self.bbox[3], self.bbox[3], self.bbox[2], self.bbox[2]]),
                                                      np.array([self.bbox[0], self.bbox[0], self.bbox[1], self.bbox[1], self.bbox[0]]))))
             if userbbox_poly.intersects(dfextents_poly):
-                extent = [np.floor(self.bbox[2]-(self.spacing/2)), np.ceil(self.bbox[-1]+(self.spacing/2)),
-                          np.floor(self.bbox[0]-(self.spacing/2)), np.ceil(self.bbox[1]+(self.spacing/2))]
+                extent = [np.floor(self.bbox[2]), np.ceil(self.bbox[-1]),np.floor(self.bbox[0]), np.ceil(self.bbox[1])]
             else:
-                print(
-                    "WARNING: User-specified bounds do not overlap with dataset bounds, proceeding with the latter.")
+                print("WARNING: User-specified bounds do not overlap with dataset bounds, proceeding with the latter.")
```

Comment:
No longer accurate

Reply:
Maybe?

Reply:
No, this is a separate issue w.r.t. my previous commit. Here, I am capturing instances where a user-specified bounding box does not intersect with the data bounds. E.g. let's say a user passed '39 40 120 121' (again south, north, west, and east respectively), but the data bounds span '41 42 122 124'. In these cases, the native data bounds are passed. 

What I just committed concerns invalid user inputs that don't correspond to legitimate coordinates. E.g. -91 south, -182 west. 

But on second though, perhaps the former case should also exit out with an exception instead of just printing a warning and proceeding with the native data bounds.

Makes sense?

Reply:
To me it would make sense to error out if I specify a bounding box that doesn't overlap with the data at all. If I specify a bounding box that is too big, then it makes sense to me that the program would shrink it down to the available data.

Reply:
Yes I agree. Will raise an exception in the cases where the user-defined bounding box does not overlap with the data bounds.

The other exception concerning user-defined bounds that supersede valid bounds remains, as that is a separate issue.